File: logic/gclient/gclient/tcpfriendserver.cpp
// modified by ket.kio.RPCGen at Mon Jan 04 14:34:52 CST 2016.

#include "tcpfriendserver.h"
#include "rpcmanagerfriendserver.h"
#include "packet.h"

namespace I3K
{

	TCPFriendServer::TCPFriendServer(KET::KIO::NetManager &managerNet)
		: KET::KIO::TCPServer<KET::KIO::SimplePacket>(managerNet)
	{
	}

	bool TCPFriendServer::DoCheckPacketType(int ptype)
	{
		switch( ptype )
		{
		// server to friend
		case Packet::eS2FPKTWhoAmI:
		case Packet::eS2FPKTCreateRole:
		case Packet::eS2FPKTSearchFriends:
		case Packet::eS2FPKTQueryFriendProp:
		case Packet::eS2FPKTUpdateFriendProp:
		case Packet::eS2FPKTSendMessageToFriend:
			return true;
		default:
			break;
		}
		return false;
	}

	void TCPFriendServer::OnOpen()
	{
		m_pRPCManagerFriendServer->OnTCPFriendServerOpen(this);
	}

	void TCPFriendServer::OnOpenFailed(KET::KIO::ErrorCode errcode)
	{
		m_pRPCManagerFriendServer->OnTCPFriendServerOpenFailed(this, errcode);
	}

	void TCPFriendServer::OnClose(KET::KIO::ErrorCode errcode)
	{
		m_pRPCManagerFriendServer->OnTCPFriendServerClose(this, errcode);
	}

	void TCPFriendServer::OnPacketSessionOpen(int sessionid, const KET::KIO::NetAddress &addrClient)
	{
		m_pRPCManagerFriendServer->OnTCPFriendServerSessionOpen(this, sessionid, addrClient);
	}

	void TCPFriendServer::OnPacketSessionClose(int sessionid, KET::KIO::ErrorCode errcode)
	{
		m_pRPCManagerFriendServer->OnTCPFriendServerSessionClose(this, sessionid, errcode);
	}

	void TCPFriendServer::OnPacketSessionRecv(int sessionid, const KET::KIO::SimplePacket *pPacket)
	{
		switch( pPacket->GetType() )
		{
		case Packet::eS2FPKTWhoAmI:
			{
				const Packet::S2F::WhoAmI *p = dynamic_cast<const Packet::S2F::WhoAmI*>(pPacket);
				m_pRPCManagerFriendServer->OnTCPFriendServerRecvWhoAmI(this, p, sessionid);
			}
			break;
		case Packet::eS2FPKTCreateRole:
			{
				const Packet::S2F::CreateRole *p = dynamic_cast<const Packet::S2F::CreateRole*>(pPacket);
				m_pRPCManagerFriendServer->OnTCPFriendServerRecvCreateRole(this, p, sessionid);
			}
			break;
		case Packet::eS2FPKTSearchFriends:
			{
				const Packet::S2F::SearchFriends *p = dynamic_cast<const Packet::S2F::SearchFriends*>(pPacket);
				m_pRPCManagerFriendServer->OnTCPFriendServerRecvSearchFriends(this, p, sessionid);
			}
			break;
		case Packet::eS2FPKTQueryFriendProp:
			{
				const Packet::S2F::QueryFriendProp *p = dynamic_cast<const Packet::S2F::QueryFriendProp*>(pPacket);
				m_pRPCManagerFriendServer->OnTCPFriendServerRecvQueryFriendProp(this, p, sessionid);
			}
			break;
		case Packet::eS2FPKTUpdateFriendProp:
			{
				const Packet::S2F::UpdateFriendProp *p = dynamic_cast<const Packet::S2F::UpdateFriendProp*>(pPacket);
				m_pRPCManagerFriendServer->OnTCPFriendServerRecvUpdateFriendProp(this, p, sessionid);
			}
			break;
		case Packet::eS2FPKTSendMessageToFriend:
			{
				const Packet::S2F::SendMessageToFriend *p = dynamic_cast<const Packet::S2F::SendMessageToFriend*>(pPacket);
				m_pRPCManagerFriendServer->OnTCPFriendServerRecvSendMessageToFriend(this, p, sessionid);
			}
			break;
		default:
			break;
		}
	}

}

File: logic/gserver/gameserver/src/i3k/friend/TCPFriendServer.java
// modified by ket.kio.RPCGen at Sat Dec 27 11:14:12 CST 2014.

package i3k.friend;

import ket.kio.PacketDecoder;
import ket.kio.PacketEncoder;
import ket.kio.NetAddress;
import ket.kio.TCPServer;
import ket.kio.SimplePacket;

import i3k.rpc.ABaseDencoder;
import i3k.rpc.Packet;

public class TCPFriendServer extends TCPServer<SimplePacket>
{

	public TCPFriendServer(RPCManagerFriendServer managerRPC)
	{
		super(managerRPC.getNetManager());
		this.managerRPC = managerRPC;
	}

	private class Dencoder extends ABaseDencoder
	{
		@Override
		public boolean doCheckPacketType(int ptype)
		{
			switch( ptype )
			{
			// server to friend
			case Packet.eS2FPKTWhoAmI:
			case Packet.eS2FPKTCreateRole:
			case Packet.eS2FPKTSearchFriends:
			case Packet.eS2FPKTQueryFriendProp:
			case Packet.eS2FPKTUpdateFriendProp:
			case Packet.eS2FPKTSendMessageToFriend:
				return true;
			default:
				break;
			}
			return false;
		}
	}

	@Override
	public PacketEncoder<SimplePacket> getEncoder()
	{
		return dencoder;
	}

	@Override
	public PacketDecoder<SimplePacket> getDecoder()
	{
		return dencoder;
	}

	@Override
	public void onOpen()
	{
		managerRPC.onTCPFriendServerOpen(this);
	}

	@Override
	public void onOpenFailed(ket.kio.ErrorCode errcode)
	{
		managerRPC.onTCPFriendServerOpenFailed(this, errcode);
	}

	@Override
	public void onClose(ket.kio.ErrorCode errcode)
	{
		managerRPC.onTCPFriendServerClose(this, errcode);
	}

	@Override
	public void onPacketSessionOpen(int sessionid, NetAddress addrClient)
	{
		managerRPC.onTCPFriendServerSessionOpen(this, sessionid, addrClient);
	}

	@Override
	public void onPacketSessionClose(int sessionid, ket.kio.ErrorCode errcode)
	{
		managerRPC.onTCPFriendServerSessionClose(this, sessionid, errcode);
	}

	@Override
	public void onPacketSessionRecv(int sessionid, SimplePacket packet)
	{
		switch( packet.getType() )
		{
		// server to friend
		case Packet.eS2FPKTWhoAmI:
			{
				Packet.S2F.WhoAmI p = (Packet.S2F.WhoAmI)packet;
				managerRPC.onTCPFriendServerRecvWhoAmI(this, p, sessionid);
			}
			break;
		case Packet.eS2FPKTCreateRole:
			{
				Packet.S2F.CreateRole p = (Packet.S2F.CreateRole)packet;
				managerRPC.onTCPFriendServerRecvCreateRole(this, p, sessionid);
			}
			break;
		case Packet.eS2FPKTSearchFriends:
			{
				Packet.S2F.SearchFriends p = (Packet.S2F.SearchFriends)packet;
				managerRPC.onTCPFriendServerRecvSearchFriends(this, p, sessionid);
			}
			break;
		case Packet.eS2FPKTQueryFriendProp:
			{
				Packet.S2F.QueryFriendProp p = (Packet.S2F.QueryFriendProp)packet;
				managerRPC.onTCPFriendServerRecvQueryFriendProp(this, p, sessionid);
			}
			break;
		case Packet.eS2FPKTUpdateFriendProp:
			{
				Packet.S2F.UpdateFriendProp p = (Packet.S2F.UpdateFriendProp)packet;
				managerRPC.onTCPFriendServerRecvUpdateFriendProp(this, p, sessionid);
			}
			break;
		case Packet.eS2FPKTSendMessageToFriend:
			{
				Packet.S2F.SendMessageToFriend p = (Packet.S2F.SendMessageToFriend)packet;
				managerRPC.onTCPFriendServerRecvSendMessageToFriend(this, p, sessionid);
			}
			break;
		default:
			break;
		}
	}

	//todo
	private Dencoder dencoder = new Dencoder();
	private RPCManagerFriendServer managerRPC;
}
